lib/widgets/compose_box.dart

```diff
@@ -257,6 +317,91 @@ class _StreamSendButtonState extends State<_StreamSendButton> {
   }
 }
 
+class _AttachFileButton extends StatelessWidget {
+  const _AttachFileButton({required this.contentController});
+
+  final ContentTextEditingController contentController;
+
+  _handlePress(BuildContext context) async {
+    FilePickerResult? result;
+    try {
+      result = await FilePicker.platform.pickFiles(allowMultiple: true, withReadStream: true);
+    } catch (e) {
+      // TODO(i18n)
+      showErrorDialog(context: context, title: 'Error', message: e.toString());
+      return;
+    }
+
+    if (result == null) {
+      return; // User cancelled; do nothing
+    }
+
+    if (context.mounted) {} // https://github.com/dart-lang/linter/issues/4007
+    else {
+      return;
+    }
+
+    final store = PerAccountStoreWidget.of(context);
+    final account = store.account;
+    final realmUrl = Uri.parse(account.realmUrl);
+
+    final List<PlatformFile> tooLargeFiles = [];
+    final List<PlatformFile> rightSizeFiles = [];
+
+    for (PlatformFile file in result.files) {
+      if ((file.size / pow(2, 20)) > store.maxFileUploadSizeMib) {
```

Comment:
I'm not sure the compiler will recognize this `pow` call as a compile-time constant. If it doesn't, that defeats optimizations it can otherwise make.

An idiomatic way to write this would be with a bitwise shift operator:
```suggestion
      if ((file.size / (1 << 20)) > store.maxFileUploadSizeMib) {
```